File: JavaScript/packages/recognizers-options/src/options/optionsRecognizer.ts
import { Recognizer, IModel, Culture } from "@microsoft/recognizers-text";
import { BooleanModel } from "./models";
import { BooleanExtractor } from "./extractors";
import { BooleanParser } from "./parsers";
import { EnglishBooleanExtractorConfiguration } from "./english/boolean";

export enum OptionsOptions {
    None = 0,
}

export default class OptionsRecognizer extends Recognizer<OptionsOptions> {
    private constructor(culture: string, options: OptionsOptions = OptionsOptions.None) {
        super(culture, options);
    }

    protected InitializeConfiguration() {
        // English models
        this.registerModel("BooleanModel", Culture.English, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new EnglishBooleanExtractorConfiguration())
        ));
    }

    getBooleanModel(): IModel {
        return this.getModel("BooleanModel");
    }
}

File: .NET/Microsoft.Recognizers.Text.Options/OptionsRecognizer.cs
using System;
using System.Collections.Generic;
using Microsoft.Recognizers.Text.Options.English.Extractors;
using Microsoft.Recognizers.Text.Options.Extractors;
using Microsoft.Recognizers.Text.Options.Models;
using Microsoft.Recognizers.Text.Options.Parsers;

namespace Microsoft.Recognizers.Text.Options
{
    public class OptionsRecognizer : Recognizer
    {
        public static readonly OptionsRecognizer Instance = new OptionsRecognizer(OptionsRecognizerOptions.None);

        public OptionsRecognizer(OptionsRecognizerOptions options)
        {
            RegisterModel(Culture.English, options.ToString(), new Dictionary<Type, IModel>
            {
                [typeof(BooleanModel)] = new BooleanModel(new BooleanParser(), new BooleanExtractor(new EnglishBooleanExtractorConfiguration()))
            });
        }

        public IModel GetBooleanModel(string culture, bool fallbackToDefaultCulture = true)
        {
            return GetModel<BooleanModel>(culture, fallbackToDefaultCulture);
        }
    }
}
